lib/notifications/display.dart

```diff
@@ -52,11 +56,28 @@ class NotificationChannelManager {
   //    settings for the channel -- like "override Do Not Disturb", or "use
   //    a different sound", or "don't pop on screen" -- their changes get
   //    reset.  So this has to be done sparingly.
-  //
-  //    If we do this, we should also look for any channel with the old
-  //    channel ID and delete it.  See zulip-mobile's `createNotificationChannel`
-  //    in android/app/src/main/java/com/zulipmobile/notifications/NotificationChannelManager.kt .
-  static Future<void> _ensureChannel() async {
+  @visibleForTesting
+  static Future<void> ensureChannel() async {
+    // See if our current-version channel already exists; delete any obsolete
+    // previous channels.
+    var found = false;
+    final channels = await _androidHost.getNotificationChannels();
+    for (final channel in channels) {
+      assert(channel != null); // TODO(#942)
+      if (channel!.id == kChannelId) {
+        found = true;
+      } else {
+        await _androidHost.deleteNotificationChannel(channel.id);
```

Comment:
It sounds like it would be helpful to mention this in the dartdoc; something like:

```dart
  /// Create our notification channel, if it doesn't already exist.
  /// 
  /// Deletes obsolete channels, if present, from old versions of the app.
```